Cleared up some confusing parts of the Access Widening docs (#671)
FabricMC/fabric-docs: develop/class-tweakers/access-widening.md

Reviewer: ```suggestion
Access widener entries start with one of three directive keywords to specify the type of modification to apply.

After the keyword come the parameters, usually the targets of the widening.

The same class, method, or field, can be targeted by multiple access widening entries, one on each line.
```

Reviewer: ```suggestion
```

Author: I feel like this is not necessary.

It's fine separated

Author: like the "including these 3 entry types" part leading to the next paragraph about entry types being able to be transitive.

Reviewer: I don't remember what this was for tbh ✌️

Author: lol

Reviewer: ```suggestion
authors:
  - An-m1654
```

Author: yay my edit is worthy enough!!!

(I usually don't put my name on my own because I feel like I'm making a small edit then plastering my name over it or something like that. I'm glad.)

## Access Directives {#access-directives}

Access widener entries start with one of three base directive keywords to specify the type of modification to apply.

After the keyword come the parameters, usually the targets of the widening.

The same class, method, or field, can be targeted by multiple access widening entries, one on each line.

> Access directives can also be made transitive by adding the `transitive-` prefix before the base access directive.

Reviewer: ```suggestion
Access directives can also be made [transitive](../class-tweakers/index#transitive-entries) by adding the `transitive-` prefix before the base access directive.
```

### Extendable {#extendable}

`extendable` can target classes and methods only:

- Classes are made public and non-final

> - Methods are made protected and non-final, even if orginally public

Reviewer 2: From my testing, this is false?

Reviewer 3: Indeed, pretty sure that would break many things

Author: After my testing, you are indeed correct. It doesn't seem to change public methods to protected. I'm going to change it.

Author: It was originally weird anyways.

Making a method extendable also makes its class extendable.

### Mutable {#mutable}

`mutable` can make a field non-final.

> To make a private final method both accessible and extendable or a private final field both accessible and mutable, you must make two separate entries in the file.

Reviewer 2: Move to `accessible` because that's what the two have in common.

Author: But extendable and mutable have to explained first before they can be brought up. So given this, it needs to be placed after extendable and mutable.

Reviewer 2: It doesn't need to, because they're all on the same page and not very far from each other.

Author: In my mind, the "make two separate entries in the file" concept is supposed to be not under accessible, extendable or mutable, but a separate thing.

I just don't know how to get out of the "scope" of mutable. It's supposed to be about ALL of them. would this be a good idea?

## Access Directives

Access widener entries start with one of three directive keywords to specify the type of modification to apply.

### Accessible

`accessible` can target classes, methods and fields:

- Fields and Classes are made public.
- Methods are made public, and final if originally private.

Making a method or field accessible also makes its class accessible.

### Extendable

`extendable` can target classes and methods:

- Classes are made public and non-final.
- Methods are made protected and non-final, and protected if originally private.
Making a method extendable also makes its class extendable.

### Mutable

`mutable` can make a field non-final.

## Using Multiple Access Directives For One Target
To make a private final field both accessible and mutable, you must make two separate entries in the file.

## Specifying Targets
For class tweaking, classes use their internal names. For fields and methods you must specify their class name, their name, and their bytecode descriptor.

...

Author: I have now changed it to look like the one above.

### Extendable {#extendable}

`extendable` can target classes and methods only:

- Classes are made public and non-final.
- Methods are made non-final, and protected if originally private.

Making a method extendable also makes its class extendable.

### Mutable {#mutable}

`mutable` can make a field non-final.

> ## Using Multiple Access Directives For One Target {#using-multiple-access-directives-for-one-target)

Reviewer 4: ```suggestion
## Using Multiple Access Directives For One Target {#using-multiple-access-directives-for-one-target}
```

Author: oh whoops

To make a private final method both accessible and extendable or a private final field both accessible and mutable, you must make two separate entries in the file.

> 

Reviewer: ```suggestion
```

## Applying Changes {#applying-changes}

To see your changes applied, you must [regenerate sources](../getting-started/generating-sources) then refresh your Gradle project. The elements you targeted should
have their access limits modified accordingly. If you had the modified class file opened, you must reopen the same class file from the decompiled source to see the

> modification. If modifications do not appear, you can try [validating the file](../class-tweakers/index#validating-the-file) and checking if any errors appear.

Reviewer: ```suggestion
To see your changes applied, you must [regenerate sources](../getting-started/generating-sources) then refresh your Gradle project. The elements you targeted should
have their access limits modified accordingly. Make sure to reopen any targeted class from the decompiled source in order to see the
modifications.

::: tip

If modifications do not appear, you can try [validating the file](../class-tweakers/index#validating-the-file) and checking if any errors appear.

:::
```

still not sure about the "regenerate sources then refresh Gradle" thing.

## Applying Changes {#applying-changes}

To see your changes applied, you must [regenerate sources](../getting-started/generating-sources) then refresh your Gradle project. The elements you targeted should
have their access limits modified accordingly. Make sure to reopen any targeted class from the decompiled source in order to see the
modifications.

::: tip

If modifications do not appear, you can try [validating the file](../class-tweakers/index#validating-the-file) and checking if any errors appear.

> :::

Reviewer: ```suggestion
:::

<!---->
```

otherwise CI will be angry

## Applying Changes {#applying-changes}

> To see your changes applied, you must [regenerate sources](../getting-started/generating-sources) then refresh your Gradle project. The elements you targeted should

Reviewer: Multiple people way smarter than me are saying, here and on Discord, that refreshing Gradle should be done before, not after, regenerating sources. I ask that you revert this change, and if you're still having that issue consider posting a message on [Github Discussions](https://github.com/orgs/FabricMC/discussions) or in the [Discord server](https://discord.fabricmc.net/).

Author: I sort of hate when people are dismissive like this, no offense. The fact is that after my own test, this does not seem to be the case. Therefore, I ask in return that you give me some time, so that I can test this with Intellij for real.

As this last part is still not yet fully confirmed, I will set this pull request to be a draft for now.

Author: maybe I overreacted :no_mouth:

Reviewer: I'm sorry if I came across as dismissive, that was not my intention. Whenever you have time to test we'll be happy to analyze your situation. But seeing that this may take a lot of discussion, screenshots, and back-and-forth, this conversation should be moved to GHD or Discord (links in the first message).

Author: This is resolved because of https://github.com/FabricMC/fabric-docs/pull/671#issuecomment-5144109608
